pkg/backend/httpstate/journal/snapshot.go

```diff
@@ -168,7 +177,10 @@ func sendBatches(
 				return
 			}
 
-			batch, results = append(batch, req.entry), append(results, req.result)
+			batch = append(batch, req.entry)
+			if req.result != nil {
+				results = append(results, req.result)
+			}
 			if cap(batch) == 0 {
```

Comment:
non-blocking: maybe fix this one too while you're in here? I know it's in the other PR as well, but it's quite simple and would be nice to get in now

```suggestion
			if len(batch) == cap(batch) {
```